src/coll/options.rs

```diff
@@ -29,58 +43,126 @@ pub enum WriteModel {
     },
 }
 
 /// Options for aggregation queries.
-#[derive(Clone, Default)]
+#[derive(Clone, Debug, Default)]
 pub struct AggregateOptions {
-    pub allow_disk_use: bool,
-    pub use_cursor: bool,
+    pub allow_disk_use: Option<bool>,
+    pub use_cursor: Option<bool>,
     pub batch_size: i32,
     pub max_time_ms: Option<i64>,
     pub read_preference: Option<ReadPreference>,
 }
 
+impl AggregateOptions {
+    pub fn new() -> Self {
+        Default::default()
+    }
+}
+
+impl From<AggregateOptions> for bson::Document {
+    fn from(options: AggregateOptions) -> Self {
+        let mut document = bson::Document::new();
+
+        if let Some(allow_disk_use) = options.allow_disk_use {
+            document.insert("allowDiskUse", Bson::Boolean(allow_disk_use));
+        }
+
+        // useCursor not currently used by the driver.
+
+
+        let cursor = doc! { "batchSize" => (options.batch_size) };
+        document.insert("cursor", Bson::Document(cursor));
+
+        // maxTimeMS is not currently used by the driver.
+
+        // read_preference is used directly by Collection::aggregate.
+
+        document
+    }
+}
+
 /// Options for count queries.
-#[derive(Clone, Default)]
+#[derive(Clone, Debug, Default)]
 pub struct CountOptions {
-    pub skip: u64,
-    pub limit: i64,
+    pub skip: Option<i64>,
+    pub limit: Option<i64>,
     pub hint: Option<String>,
     pub hint_doc: Option<bson::Document>,
     pub max_time_ms: Option<i64>,
     pub read_preference: Option<ReadPreference>,
 }
 
+impl CountOptions {
+    pub fn new() -> Self {
+        Default::default()
+    }
+}
+
+impl From<CountOptions> for bson::Document {
+    fn from(options: CountOptions) -> Self {
+        let mut document = bson::Document::new();
+
+        if let Some(skip) = options.skip {
+            document.insert("skip", Bson::I64(skip));
+        }
+
+        if let Some(limit) = options.limit {
+            document.insert("limit", Bson::I64(limit));
+        }
+
+        if let Some(hint) = options.hint {
+            document.insert("hint", Bson::String(hint));
+        }
+
+        if let Some(hint_doc) = options.hint_doc {
+            document.insert("hint_doc", Bson::Document(hint_doc));
+        }
+
+        // maxTimeMS is not currently used by the driver.
+
+        // read_preference is used directly by Collection::count.
+
+        document
+    }
+}
+
 /// Options for distinct queries.
-#[derive(Clone, Default)]
+#[derive(Clone, Debug, Default)]
 pub struct DistinctOptions {
     pub max_time_ms: Option<i64>,
     pub read_preference: Option<ReadPreference>,
 }
 
+impl DistinctOptions {
+    pub fn new() -> Self {
+        Default::default()
+    }
+}
+
 /// Options for collection queries.
-#[derive(Clone)]
+#[derive(Clone, Debug, Default)]
 pub struct FindOptions {
     pub allow_partial_results: bool,
     pub no_cursor_timeout: bool,
-    pub op_log_replay: bool,
-    pub skip: u32,
-    pub limit: i32,
+    pub oplog_replay: bool,
```

Comment:
In MongoDB parlance, "oplog" is generally used as a single word, despite the fact that it isn't really a word at all in actual parlance